### PR TITLE
HPC-GAP: sync hpcgap/src/gvars.h; partial sync of hpcgap/src/gvars.c

### PR DESCRIPTION
**Pull request by fingolfin**
This PR gets rid of the last header file in `hpcgap/src`.
It also hides implementation details of gvars, which will make it easier to refactor gvars in the future.
Note that calling the function `ValGVar` outside of `gvars.c` is slower than using the former `VAL_GVAR` macro. However, I looked at all places we use(d) `VAL_GVAR`, and I am pretty sure they are not time critical.

### CODE REVIEW

File: hpcgap/src/gvars.c [hunk 1]
```diff
@@ -146,11 +160,16 @@ void UnlockGVars() {
 **  will return the value of <gvar>  after evaluating <gvar>-s expression, or
 **  0 if <gvar> was not an automatic variable.
 **
-**  'VAL_GVAR' is defined in the declaration part of this package as follows
-**
-#define VAL_GVAR(gvar)          (PtrGVars[GVAR_BUCKET(gvar)] \
-                                [GVAR_INDEX(gvar)-1])
 */
+#undef VAL_GVAR
+#define VAL_GVAR(gvar)          (PtrGVars[GVAR_BUCKET(gvar)] \
```
Comment (fingolfin) [inline hpcgap/src/gvars.c]: `VAL_GVAR` and `ValGVar` are not the same in HPC-GAP -- the latter contains a `MEMBAR_READ()`.

So perhaps what you suggest is the right thing to do -- but I did not feel competent to quickly judge this, and hence decided to defer this until a latter stage; I'd rather not put a change into a refactoring commit which might have major consequences.

If it is the #undef/redefine dance you dislike (I actually do, too): I could rename the "second" `VAL_GVAR` defined here to, say `VAL_GVAR_RAW` or `VAL_GVAR_INTERN`, and perhaps add a FIXME explaining that somebody should look into whether we need it.... Or perhaps you have another suggestion?

Comment (ChrisJefferson) [inline hpcgap/src/gvars.c]: Yes, it was more the redefine, now VAL_GVAR means something different in this file, and outside it, so people looking at the definition have to be careful.

Comment (fingolfin) [inline hpcgap/src/gvars.c]: OK, that's a good point. I'll take care of it.

File: hpcgap/src/gvars.c [hunk 2]
```diff
@@ -146,12 +160,21 @@ void UnlockGVars() {
...
-**
-#define VAL_GVAR(gvar)          (PtrGVars[GVAR_BUCKET(gvar)] \
-                                [GVAR_INDEX(gvar)-1])
 */
 
+// FIXME/TODO: Do we still need the VAL_GVAR_INTERN macro, or can we replace
+// it by ValGVar everywhere? The difference is of course the memory barrier,
+// which might cause a performance penalty (OTOH, not using it right now might
+// or might not be a bug?!?)
+#define VAL_GVAR_INTERN(gvar)          (PtrGVars[GVAR_BUCKET(gvar)] \
```
Comment (fingolfin) [inline hpcgap/src/gvars.c]: @rbehrends Could you perhaps comment on this?

**Pull request merged**